pandas/core/arrays/numpy_.py

```diff
@@ -276,6 +276,15 @@ def __setitem__(self, key, value) -> None:
 
         self._ndarray[key] = value
 
+    def astype(self, dtype, copy: bool = True) -> ArrayLike:
+        if dtype == self.dtype:
```

Comment:
So the problem is that `np.dtype(object) == ExtensionDtype` raises an error, which is an error coming from numpy and not directly solvable in pandas. 
So to workaround it, can you do `is_dtype_equal(dtype, self.dtype):`, with `is_dtype_equal` imported from `pandas.core.dtypes.common`

Reply:
Fixed and moved checks to base class. This just calls super().astype now.